### PR TITLE
Added current weapon ammunition count displaying.

### PR DESCRIPTION
**Pull request by RICCIARDI-Adrien**

### CODE REVIEW

File: src/state_control/state_control_Lara.cpp [hunk 1]
```diff
@@ -3326,27 +3332,35 @@ int StateControl_LaraDoTwoHandWeaponFrame(struct entity_s *ent, struct ss_animat
         {
             case ITEM_SHOTGUN:
                 weapon = getShotgun();
+                ent->character->currentAmmunitionInventoryItemId = ITEM_SHOTGUN_AMMO; // TODO multiple ammo in TR4
```
Comment (TeslaRus) [inline src/state_control/state_control_Lara.cpp]: ITEM_SHOTGUN_AMMO1, 2, 3, 4... and it will be fixed ))

File: src/weapons.cpp [hunk 2]
```diff
@@ -463,30 +463,29 @@ bool Weapons_GetIdsFromActionKey(int weaponId, int *pointerModelId, int *pointer
...
     }
     else
     {
         *pointerModelId = tombRaider5WeaponIds[weaponId].modelId;
         *pointerInventoryItemId = tombRaider5WeaponIds[weaponId].inventoryItemId;
-        return true;
     }
+
+    // Is a weapon bound to this key ?
+    if (*pointerModelId == 0) return false;
```
Comment (TeslaRus) [inline src/weapons.cpp]: return 0 !=  *pointerModelId;

File: src/gui/gui.cpp [hunk 3]
```diff
@@ -47,6 +47,22 @@ static GLfloat screenSize[2];
...
+    int ammunitionInventoryItemId, ammunitionCount;
+
+    // Retrieve currently used ammunition
+    entity_p player = World_GetPlayer();
+    ammunitionInventoryItemId = player->character->currentAmmunitionInventoryItemId;
+    if((ammunitionInventoryItemId == ITEM_PISTOL_AMMO) || (ammunitionInventoryItemId == ITEM_GRAPPLEGUN_AMMO)) return; // Do not display ammunition amount for unlimited ammunition weapons
+
+    // Display ammunition count to top right screen corner
+    ammunitionCount = Inventory_GetItemsCount(player->inventory, ammunitionInventoryItemId);
+    sprintf(ammunitionsCountString, "%d", ammunitionCount);
```
Comment (TeslaRus) [inline src/gui/gui.cpp]: snprintf - less compiler warnings

File: src/character_controller.h [hunk 4]
```diff
@@ -285,6 +285,7 @@ typedef struct character_s
     uint16_t                    bone_r_hand_end;
     int16_t                     weapon_id;
     int16_t                     weapon_id_req;
+    int                         currentAmmunitionInventoryItemId;
```
Comment (TeslaRus) [inline src/character_controller.h]: naming code style... low_case

**Comment by TeslaRus** [comment]
thanks for PR!
will be merged after fixes

File: src/state_control/state_control_Lara.cpp (hunk 1 above)
Comment (RICCIARDI-Adrien) [reply src/state_control/state_control_Lara.cpp]: ITEM_SHOTGUN_NORMAL_AMMO, ITEM_SHOTGUN_WIDESHOT_AMMO and others are existing yet, but I think that there is no way to select them for now.

File: src/weapons.cpp (hunk 2 above)
Comment (RICCIARDI-Adrien) [reply src/weapons.cpp]: Ok, I will change this in another commit.

File: src/gui/gui.cpp (hunk 3 above)
Comment (RICCIARDI-Adrien) [reply src/gui/gui.cpp]: Ok, I will change this in another commit.

File: src/character_controller.h (hunk 4 above)
Comment (RICCIARDI-Adrien) [reply src/character_controller.h]: Ok, I will change this in another commit.

File: src/weapons.cpp (hunk 2 above)
Comment (RICCIARDI-Adrien) [reply src/weapons.cpp]: I didn't read well your proposal. I prefer mine as I find it more readable. However, I can change for yours if you want to.

**Comment by RICCIARDI-Adrien** [comment]
Thank you !
Pull request is ready for merging.

File: src/game.cpp [hunk 5]
```diff
@@ -850,6 +850,15 @@ void Game_Prepare()
         engine_camera.transform.M4x4[12 + 2] += player->character->height;
         engine_camera.transform.angles[0] = player->transform.angles[0] + 180.0f;
         engine_camera.current_room = player->self->room;
+
+        // Equip pistols if they are available in inventory
+        if(Inventory_GetItemsCount(player->inventory, ITEM_PISTOL) > 0)
```
Comment (TeslaRus) [inline src/game.cpp]: todo: (for myself, do not forget) check player->character->set_weapon_model_func != NULL

File: src/weapons.cpp (hunk 2 above)
Comment (TeslaRus) [reply src/weapons.cpp]: too much strings, mine will be more readable ;-)

File: src/gui/gui.cpp [hunk 6]
```diff
@@ -47,6 +47,22 @@ static GLfloat screenSize[2];
...
 static void Gui_FillBackgroundBuffer();
 
+static void Gui_DrawAmmunitionCount()
+{
+    char ammunitionsCountString[11]; // Should have room enough to store a 10-digit number and the string terminating zero
+    int ammunitionInventoryItemId, ammunitionCount;
+
+    // Retrieve currently used ammunition
+    entity_p player = World_GetPlayer();
+    ammunitionInventoryItemId = player->character->current_ammunition_inventory_item_id;
```
Comment (vvs-) [inline src/gui/gui.cpp]: Program terminated with signal SIGSEGV, Segmentation fault.
#0  Gui_DrawAmmunitionCount () at /root/1/games/opentomb/src/gui/gui.cpp:57
57	    ammunitionInventoryItemId = player->character->current_ammunition_inventory_item_id;

Comment (TeslaRus) [reply src/gui/gui.cpp]: fixed

**Pull request merged**